process/dns_v1.go

```diff
@@ -0,0 +1,336 @@
+package process
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math"
+
+	"github.com/DataDog/mmh3"
+)
+
+// DNS data is encoded as a very basic bucketed hash table.  There are three blocks, or buffers, of data:
+//
+//	The "name" block is all of the unique DNS names.  The length of the name is stored as a varint
+//	followed by the name itself
+//
+//	The "bucket" block contains all of the hash buckets.  The format of each bucket is:
+//		varint for number of entries in bucket
+//		For each entry in the bucket:
+//			varint for length of ip
+//			ip bytes
+//			varint for number of names associated with the ip
+//			Each associated name is encoded as a varint which is the position of the actual name string in the name block
+//
+//	The "position" block is a list of varints, one for each bucket, where each varint is a pointer to the start
+//	of the bucket in the bucket block
+//
+// The overall buffer is encoded as:
+//	1 byte indicating version
+// 	1 byte indicating the number of buckets
+//	varint indicating the length of the name buffer.
+//	varint indicating the length of the position buffer
+// 	varint indicating the position of the "middle" (bucketCount / 2) bucket in the position block
+//		We will use this to skip the half of the buckets when searching for the target bucket index
```

Comment:
Does the  `get`  operation always need to re-process position block ? If this optimization is necessary, maybe we should add a `decoder` interface that can parse the position block once and save the result.

Reply:
>  Does the get operation always need to re-process position block ? 

Yes.  The pathological case is that you would have to read `bucketCount/2` varints if the target bucket is either `bucketCount` or `(bucketCount/2)-1`  

> maybe we should add a decoder interface that can parse the position block once and save the result.

We could do that but we would lose the "sugar" of adding `CollectorConnections.GetDNS`.  The read path is already slightly (~11%) faster than a proto map.  Do you think making it faster would be a worthwhile tradeoff? I don't have strong feelings here.

Reply:
I don't feel strongly about merging this with or without a stateful decoder. We can always add that later.